File: server/internal/dto/template.go
package dto

// TemplateView is the response for a template record.
type TemplateView struct {
	ID           string      `json:"id"`
	RepoID       string      `json:"repoId"`
	SerialNo     string      `json:"serialNo"`
	Name         string      `json:"name"`
	QuestionType string      `json:"questionType"`
	TemplateData interface{} `json:"template"`
	Mode         string      `json:"mode"`
	Category     string      `json:"category"`
	Tag          string      `json:"tag"`
	Priority     *int        `json:"priority"`
	PreviewURL   string      `json:"previewUrl"`
	Shared       *bool       `json:"shared"`
}

// TemplateQuery is used to query templates.
type TemplateQuery struct {
	PageRequest
	ID           string `form:"id"`
	RepoID       string `form:"repoId"`
	Name         string `form:"name"`
	Category     string `form:"category"`
	Mode         string `form:"mode"`
	QuestionType string `form:"questionType"`
}

// TemplateRequest is used to create/update/delete a template.
type TemplateRequest struct {
	ID           string      `json:"id"`
	RepoID       string      `json:"repoId"`
	Name         string      `json:"name"`
	QuestionType string      `json:"questionType"`
	TemplateData interface{} `json:"template"`
	Mode         string      `json:"mode"`
	Category     string      `json:"category"`
	Tag          string      `json:"tag"`
	Priority     *int        `json:"priority"`
	PreviewURL   string      `json:"previewUrl"`
	Shared       *bool       `json:"shared"`
}

// CategoryQuery is used to query template categories.
type CategoryQuery struct {
	Mode string `form:"mode"`
}

// TagQuery is used to query template tags.
type TagQuery struct {
	Mode     string `form:"mode"`
	Category string `form:"category"`
}

File: website/src/types/template.ts
// Template query parameters
export interface TemplateQuery {
  pageIndex?: number
  pageSize?: number
  id?: string
  repoId?: string
  name?: string
  category?: string
  mode?: string
  questionType?: string
}

// Template view response
export interface TemplateView {
  id: string
  repoId: string
  serialNo: string
  name: string
  questionType: string
  template: object
  mode: string
  category: string
  tag: string
  priority: number | null
  previewUrl: string
  shared: boolean | null
}

// Template create/update request
export interface TemplateRequest {
  id?: string
  repoId?: string
  name: string
  questionType?: string
  template?: object
  mode?: string
  category?: string
  tag?: string
  priority?: number
  previewUrl?: string
  shared?: boolean
}

// Category query
export interface CategoryQuery {
  mode?: string
}

// Category view
export interface CategoryView {
  id: string
  name: string
}

// Tag query
export interface TagQuery {
  mode?: string
  category?: string
}

// Tag view
export interface TagView {
  id: string
  name: string
}
